service/controller/app/resource/chart/desired.go

```diff
@@ -85,28 +89,42 @@ func generateAnnotations(input map[string]string) map[string]string {
 	return annotations
 }
 
-func generateConfig(cr v1alpha1.App, appCatalog v1alpha1.AppCatalog, chartNamespace string) v1alpha1.ChartSpecConfig {
+func (r *Resource) generateConfig(ctx context.Context, cr v1alpha1.App, appCatalog v1alpha1.AppCatalog, chartNamespace string) (v1alpha1.ChartSpecConfig, error) {
```

Comment:
This function needs k8sclient from now since it has to fetch `resourceVersion` from configmap/secret objects.

Reply:
For TC app CRs the k8s client is for the tenant. So we need to use the client from the controller context.